spec/cli/navigate_spec.rb

```diff
@@ -228,6 +228,33 @@ def require_relative(path)
     end
   end
 
+  context 'when the collections endpoint has a depth parameter', :vcr do
+    it "should not be exposed to the user" do
+      nav = Razor::CLI::Parse.new(%w[-u http://fred:dead@localhost:8150/api brokers --depth 1]).navigate
+      expect {nav.get_document}.to raise_error(OptionParser::InvalidOption, 'invalid option: --depth')
+    end
+
+    it "should be set to 1 when the endpoint is the final query" do
+      Razor::CLI::Parse.new(['register-node', '--installed', 'true', '--hw-info', 'net0=78:31:c1:be:c8:01']).
+          navigate.get_document
+
+      nav = Razor::CLI::Parse.new(['-u', 'http://fred:dead@localhost:8150/api', 'nodes']).navigate
+      nav.get_document
+      expect(nav.doc_resource.url).to match(/nodes\?depth=1/)
+    end
+
+    it "should not carry-over to later requests in a nested query" do
+      broker_name = "test_broker"
+      Razor::CLI::Parse.new([
+        'create-broker', '--name', "#{broker_name}", '-c', 'server=puppet.example.org', '-c', 'environment=production', '--broker-type', 'puppet']
+      ).navigate.get_document['name']
+
+      nav = Razor::CLI::Parse.new(['-u', 'http://fred:dead@localhost:8150/api', 'brokers', "#{broker_name}"]).navigate
+      nav.get_document
+      expect(nav.doc_resource.url).not_to match(/depth/)
```

Comment:
I think it would be useful (and use a trivial amount of memory) for us to cache a list of the URLs that our navigation hits, for use in spec tests. That way we can verify these, including arguments passed:
- http://\<url>/api
- http://\<url>/api/collections/brokers
- http://\<url>/api/collections/brokers/\<name>

Reply:
Talked about this on HipChat, but a good way we can do this is to extend the Nav class in spec_helper to implement the caching feature, and then mock out Nav.new to return our derived class. This separates test-specific code from app-specific code.

We also agreed this should be done separately, not a part of this PR.

Reply:
The best intermediate step I could see is putting in a `# todo` as a note to also check `http://<url>/api/collections/brokers` for the depth, namely to ensure it doesn't equal 1.

Reply:
Done.